support/spray/src/test/scala/sjsonnew/support/spray/CalendarFormatsSpec.scala

```diff
@@ -20,16 +20,59 @@ package support.spray
 import spray.json.{ JsValue, JsNumber, JsString, JsNull, JsTrue, JsFalse, JsObject }
 import org.specs2.mutable._
 import java.util.{ Calendar, TimeZone }
+import java.time._
 
 class CalendarFormatsSpec extends Specification with BasicJsonProtocol {
   "The dateStringIso" should {
+    // JDK 8 / Joda dates
+    val zdt = ZonedDateTime.of(1999, 1, 1, 0, 0, 0, 0, ZoneId.of("UTC"))
+    val millis = zdt.plusNanos(1000 * 1000)
+    // zoneless instant in time
+    val inst = millis.toInstant
+    val ld = LocalDate.of(1999, 1, 1)
+
+    // legacy dates
     val seconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     seconds.clear
     seconds.set(1999, 1, 1, 0, 0, 0)
+    
     val milliseconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     milliseconds.clear
     milliseconds.set(1999, 1, 1, 0, 0, 0)
     milliseconds.set(Calendar.MILLISECOND, 1)
+
+    "convert a ZonedDateTime to JsString" in {
+      Converter.toJsonUnsafe(zdt) mustEqual JsString("1999-01-01T00:00:00Z")
+    }
+
+    "convert the JsString back to the ZonedDateTime" in {
+      Converter.fromJsonUnsafe[ZonedDateTime](JsString("1999-01-01T00:00:00Z")).toInstant mustEqual zdt.toInstant
```

Comment:
Oh yeah, joda-time/java.time's DateTime/ZonedDateTime (and similar) hold more information that ISO 8601, namely the timezone id rather than just its offset. So you can distinguish if the time is Europe/Paris rather than Europe/Rome.

I actually had to work around this lossyness back in 2013 when persisting and retrieving dates with Hibernate. Possibly one of my first OSS contributions, or at least one of the first good ones: https://github.com/JadiraOrg/jadira/pull/3/files